### DISCUSSION TITLE
Cubemap creation

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#6674] [Contribute]
### Is your feature request related to a problem? Please describe.

After reading this discussion https://github.com/ImageMagick/ImageMagick/discussions/1988 
it seems like it is not possible to create a dds or ktx cubemap from 6 separate images. The dds plugin for Gimp has a "save as cubemap" option to do this, but Gimp is not easy to integrate in an automated texture pipeline.

### Describe the solution you'd like

Would be really nice to have a dds option for this kind of layered cubemap textures.
The call could look like
magick "x.png" "-x.png" "y.png" "-y.png" "z.png" "-z.png" -define dds:cubemap=true cubemap.dds

**Comment by fmw42** [comment]
Don't know if this is helpful, but I have a cube2sphericalpano bash unix script at http://www.fmwconcepts.com/imagemagick/cube2sphericalpano/index.php

**Comment by fmw42** [comment]
I do not know if this is helpful, but I have bash unix shell scripts for Imagemagick for going both ways at http://www.fmwconcepts.com/imagemagick/cube2sphericalpano/index.php and http://www.fmwconcepts.com/imagemagick/sphericalpano2cube/index.php

**Comment by ghost** [comment]
Thanks @fmw42 , that's interesting, but not what I'm looking for. I need a dxt compressed texture where each cube face is a separate layer

**Comment by fmw42** [comment]
Perhaps IM developers can take my script and then write out in DXT format?

**Comment by fmw42** [comment]
Is this the structure you need for your six images:

![Skybox_example](https://github.com/ImageMagick/ImageMagick/assets/13458218/d211f09d-45ea-459a-86c6-635bbe246f47)

If so, you can create this with +append and -append. (Or possibly just montage using null: for place holders).
Then IM has a define for writing dds output in dxt compression. See the formats page or defines page for -define dds:compression={dxt1, dxt5, none})
